plugins/browser-plugin-accelerator-ecommerce/src/types.ts

```diff
@@ -0,0 +1,230 @@
+   * E.g. product list, search results, shop the look, frequently bought with
+   */
+  name?: string;
+}
+
+/**
+ * Type/Schema for a checkout_step event in Ecommerce
+ */
+export interface CheckoutStep {
+  /* Checkout step index */
+  step: number;
+  /* Shipping address postcode */
+  shipping_postcode?: string;
+  /* Billing address postcode */
+  billing_postcode?: string;
+  /* Full shipping address */
+  shipping_full_address?: string;
+  /* Full billing address */
+  billing_full_address?: string;
+  /* Can be used to discern delivery providers DHL, PostNL etc. */
+  delivery_provider?: string;
+  /* Store pickup, standard delivery, express delivery, international */
+  delivery_method?: string;
+  /* Coupon applied at checkout */
+  coupon_code?: string;
+  /* Selection of 'existing user' or 'guest checkout' */
+  account_type?: string;
+  /* Any kind of payment method the user selected to proceed. Card, PayPal, Alipay etc. */
+  payment_method?: string;
+  /* Invoice or receipt */
+  proof_of_payment?: string;
+  /* If opted in to marketing campaigns to the email address */
+  marketing_opt_in?: boolean;
+}
+
+/**
+ * Type for a cart entity in Ecommerce
+ */
+export interface Cart {
+  /**
+   * The unique ID representing this cart
+   */
+  cart_id?: string;
+  /**
+   * The total value of the cart after this interaction
+   */
+  total_value: number;
+  /**
+   * The currency used for this cart (ISO 4217)
+   */
+  currency: string;
+  /**
+   * Array of products added/removed from cart
+   */
+  products?: Product[];
+}
+
+/**
+ * Type/Schema for a page entity in Ecommerce
+ */
+export interface Page {
+  /**
+   * The type of the page that was visited (e.g. homepage, product page, cart,checkout page, etc)
+   */
+  type: string;
+  /**
+   * The language that the web page is based in
+   */
+  language?: string;
+  /**
+   * The locale version of the site that is running
+   */
+  locale?: string;
+}
+
+/**
+ * Type/Schema for a product entity in Ecommerce
+ */
+export type Product = {
+  /**
+   * The SKU or product ID
+   */
+  id: string;
+  /**
+   * The name or title of the product
+   */
+  name?: string;
+  /**
+   * The category the product belongs to.
+   * Use a consistent separator to express multiple levels. E.g. Woman/Shoes/Sneakers
+   */
+  category?: string;
+  /**
+   * The price of the product at the current time.
+   */
+  price: number;
+  /**
+   * The recommended or list price of a product
+   */
+  list_price?: number;
+  /**
+   * The quantity of the product taking part in the action. Used for Cart events.
+   */
+  quantity?: number;
+  /**
+   * The size of the product
+   */
+  size?: string | null;
+  /**
+   * The variant of the product
+   */
+  variant?: string | null;
+  /**
+   * The brand of the product
+   */
+  brand?: string | null;
+  /**
+   * The inventory status of the product (e.g. in stock, out of stock, preorder, backorder, etc)
+   */
+  inventory_status?: string | null;
```

Comment:
Why do some of the optional properties have the `null` as an accepted type while others like `creative_id` don't?

Reply:
Fixed btw